pliers/filters/audio.py

```diff
@@ -13,3 +15,39 @@ class AudioFilter(Filter):
 
 class AudioTrimmingFilter(TemporalTrimmingFilter, AudioFilter):
     pass
+
+
+class AudioResamplingFilter(AudioFilter):
+    
+    ''' Librosa-based audio resampling Filter.
+        Uses librosa.core.resample function.
+
+    Args:
+        target_sr (float): Target sampling rate (in Hz).
+        resample_type (str): Type of resampling. Must be one of 
+            'kaiser_best', 'kaiser_fast', 'scipy', 'fft' or 
+            'polyphase'. See librosa.core.resample documentation for 
+            more details.
+        librosa_kwargs: Optional keyword args passed onto the
+            librosa resampling function.
+    '''
+    
+    _log_attributes = ('target_sr', 'resample_type')
+    
+    def __init__(self, target_sr=22050, resample_type='kaiser_best',
+                 **librosa_kwargs):
+        verify_dependencies(['librosa'])
+        self.target_sr = target_sr
+        self.resample_type = resample_type
+        self.librosa_kwargs = librosa_kwargs
+        super(AudioResamplingFilter, self).__init__()
+
+    def _filter(self, stim):
+        stim.data = librosa.core.resample(y=stim.data,
```

Comment:
Not sure if librosa's `resample` already does this (doesn't look like it, at a glance), but it might make sense to first check the current sampling rate and only call librosa if the `stim.sampling_rate` is different from the `target_sr`. If the two match, we can just return the existing stim—no need to resample, which is lossy and/or time-consuming.